lib/cli/input/init.js

```diff
@@ -1,32 +1,26 @@
 const inquirer = require('inquirer');
+const Prompt = require('../../utils/Prompt');
 
-const credentials = [
-{
-  name: 'username',
-  type: 'input',
-  message: 'Enter your AutolabJS username:',
-  validate: function(value) {
-    if (value.length) {
-      return true;
-    }
-    else {
-      return 'Please enter your username';
-    }
-  }
-},
-{
-  name: 'password',
-  type: 'password',
-  message: 'Enter your password:',
-  validate: function(value) {
+const getLengthValidator = (invalidMessage) => {
+  return (value) => {
     if (value.length) {
       return true;
     }
     else {
-      return 'Please enter your password';
+      return invalidMessage;
```

Comment:
This return message does not seem consistent with the previous approach of giving the prompt again. Am I missing something in the code?

Reply:
The validate function is the same for both the prompts except for the message displayed when invalid. Hence I'm passing this message as a parameter.

Reply:
@1AmNegan Good use of closures. Nice code.